docs/compactor.md

```diff
@@ -0,0 +1,93 @@
+
+**How the solution works:** The newly introduced compact command does not disturb the running ETCD while capturing the compacted snapshot. The command is designed to run potentially separately (from the main ETCD process/container/pod). ETCD Druid may take care of starting the newly introduced compact command as a separate job (scheduled periodically and also on-demand if required) based on some user defined parameter.
+
+### **Points to take care while saving the compacted snapshot:**
+As compacted snapshot and the existing periodic full snapshots are taken by different processes running in different pods but accessing same store to save the snapshots, some problems may arise:
+1.	When uploading the compacted snapshot to the snapstore, there is the problem of how does the restorer know when to start using the newly compacted snapshot. This communication needs to be atomic.
+2.	With a regular schedule for compaction that happens potentially separately from the main etcd pod, is there a need for regular scheduled full snapshots anymore?
+
+#### **How to swap full snapshot with compacted snapshot atomically**
+
+Currently, full snapshots and the subsequent delta snapshots are grouped under same prefix path in the snapstore. When a full snapshot is created, it is placed under a prefix/directory with the name comprising of timestamp. Then subsequent delta snapshots are also pushed into the same directory. Thus each prefix/directory contains a single full snapshot and the subsequent delta snapshots. So far, it is the job of ETCDBR to start main ETCD process and snapshotter process which takes full snapshot and delta snapshot periodically. But as per our proposal, compaction will be running as parallel process to main ETCD process and snapshotter process. So we can't reliably co-ordinate between the processes to achieve switching to the compacted snapshot as the base snapshot atomically.```
+
+**Current Directory Structure**
+```yaml
+- Backup-192345
+    - Full-Snapshot-192345
+    - Incremental-Snapshot-192355
+    - Incremental-Snapshot-192365
+    - Incremental-Snapshot-192375
+- Backup-192789
+    - Full-Snapshot-192789
+    - Incremental-Snapshot-192799
+    - Incremental-Snapshot-192809
+    - Incremental-Snapshot-192819
+```
+
+To solve the problem, proposal is:
+1. ETCDBR will take the first full snapshot after it starts main ETCD Process and snapshotter process. But after taking the first full snapshot, snapshotter will only continue taking delta snapshots but no further full snapshots.
```

Comment:
This is not a central point of the design and is not yet implemented either (we decided to postpone this implementation to after observing the backup compaction at work in the landscapes). Can you please move this to a later sections (say, something titled `Future Work`, for example)?